### PR TITLE
Add methods to get machine tasks - aka machine deployment history

### PR DESCRIPTION
**Pull request by matt-richardson**

### CODE REVIEW

File: source/Octopus.Client.Tests/Integration/OctopusClient/OctopusAsyncTestClient.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+using System.Net;
+using System.Reflection;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Octopus.Client.Model;
+using Octopus.Client.Serialization;
+
+namespace Octopus.Client.Tests.Integration.OctopusClient
+{
+    public class OctopusAsyncTestClient : OctopusAsyncClient
```
Comment (droyad) [inline source/Octopus.Client.Tests/Integration/OctopusClient/OctopusAsyncTestClient.cs]: The repository code looks good. I'm not convinced that the `OctopusTestClient` provides any value over substituting `IOctopusClient`

File: source/Octopus.Client/Repositories/Async/MachineRepository.cs [hunk 2]
```diff
@@ -51,6 +52,21 @@ public Task<List<MachineResource>> FindByThumbprint(string thumbprint)
             return Client.Get<List<MachineResource>>(Client.RootDocument.Link("machines"), new { id = "all", thumbprint });
         }
 
+        public async Task<List<TaskResource>> GetTasks(MachineResource machine, int skip = 0)
```
Comment (droyad) [inline source/Octopus.Client/Repositories/Async/MachineRepository.cs]: This is a bit strange, it gets all pages, but allow you to skip the first few.

File: source/Octopus.Client/Repositories/Async/MachineRepository.cs [hunk 3]
```diff
@@ -51,6 +52,21 @@ public Task<List<MachineResource>> FindByThumbprint(string thumbprint)
...
+        {
+            if (machine == null) throw new ArgumentNullException(nameof(machine));
+
+            var resources = new List<TaskResource>();
+
+            await Client.Paginate<TaskResource>(machine.Link("TasksTemplate"), new { skip }, page =>
```
Comment (droyad) [inline source/Octopus.Client/Repositories/Async/MachineRepository.cs]: This should use `Client.ListAll`

File: source/Octopus.Client/Repositories/Async/MachineRepository.cs (hunk 2 above)
Comment (matt-richardson) [inline source/Octopus.Client/Repositories/Async/MachineRepository.cs]: Yeah. On reflection, probably not necessary.

File: source/Octopus.Client/Repositories/Async/MachineRepository.cs (hunk 3 above)
Comment (matt-richardson) [inline source/Octopus.Client/Repositories/Async/MachineRepository.cs]: Ahh - thats the magic I was looking for.

File: source/Octopus.Client.Tests/Integration/OctopusClient/OctopusAsyncTestClient.cs (hunk 1 above)
Comment (matt-richardson) [inline source/Octopus.Client.Tests/Integration/OctopusClient/OctopusAsyncTestClient.cs]: I had a few issues where it wasn't deserialising to the correct type, so I wanted to make sure it was testing the whole thing. The cycle time from make the code change, run the tests, run the build, pull the nupkg into the API tests, update the API tests, run the API tests, find the deserialisation error, repeat... was pretty high, so wanted to make sure it was tested locally.

**Pull request merged**